src/auth/auth.service.ts

```diff
@@ -19,24 +22,94 @@ export class AuthService {
   }
 
   async authenticateGoogle(token: string) {
-    const tokenInfo = await this.oauthClient.getTokenInfo(token);
+    const { email } = await this.oauthClient.getTokenInfo(token);
+    const user = await this.userService.getUser(email);
```

Comment:
Maybe it will be better to extract auth logic into another function? It will be easier to add any other oauth providers